src/core/services/aggregation.js

```diff
@@ -1,34 +1,66 @@
+import * as _ from './utility';
+
 export default class Aggregation {
 	constructor() {
 	}
 
 	static first(rows, getValue) {
-		if(!rows.length){
+		if (!rows.length) {
 			return null;
 		}
 
 		return getValue(rows[0]);
 	}
 
 	static last(rows, getValue) {
-		if(!rows.length){
+		if (!rows.length) {
 			return null;
 		}
 
 		return getValue(rows[rows.length - 1]);
 	}
 
 	static max(rows, getValue) {
-		if(!rows.length){
+		if (!rows.length) {
 			return null;
 		}
 
-		var max = Math.max.apply(Math, rows.map(getValue));
-		return max;
+		return getValue(_.maxBy(rows, d => getValue(d)));
+	}
+
+	static min(rows, getValue) {
+		if (!rows.length) {
+			return null;
+		}
+
+		return getValue(_.minBy(rows, d => getValue(d)));
+	}
+
+	static minMax(rows, getValue) {
+		return [this.min(rows, getValue), this.max(rows, getValue)];
+	}
+
+	static avg(rows, getValue) {
+		const length = rows.length;
+		return rows.reduce(function (reduced, d) {
+			return reduced + getValue(d) / length;
+		}, 0);
+	}
+
+	static sum(rows, getValue) {
+		return rows.reduce(function (reduced, d) {
+			return reduced + Number(getValue(d));
+		}, 0);
```

Comment:
On sum Number needed (because without it sum concat strings to one), and on avg it's nod need because automatic cast occured